base/linalg/tridiag.jl

```diff
@@ -149,35 +149,43 @@ eigvecs{T<:BlasFloat,Eigenvalue<:Real}(A::SymTridiagonal{T}, eigvals::Vector{Eig
 istriu(M::SymTridiagonal) = all(M.ev .== 0)
 istril(M::SymTridiagonal) = all(M.ev .== 0)
 
-function tril(M::SymTridiagonal, k::Integer=0)
+function tril!(M::SymTridiagonal, k::Integer=0)
     n = length(M.dv)
     if abs(k) > n
         throw(ArgumentError("requested diagonal, $k, out of bounds in matrix of size ($n,$n)"))
     elseif k < -1
-        return SymTridiagonal(zeros(M.dv),zeros(M.ev))
+        fill!(M.ev,0)
+        fill!(M.dv,0)
+        return Tridiagonal(M.ev,M.dv,M.ev)
```

Comment:
I think one of the off-diagonals should be copied or still use `zeros`, otherwise you have a secretly-symmetric object on your hands.